docs/faas/aws-lambda.md

```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
-
-Example pseudo implementation:
+This propagator is expected to replace the `xray` propagator in the `OTEL_PROPAGATORS` list. The behavior for this propagator is described in pseudo code below.
 
 ```
 extract(context, carrier) {
+    // To avoid potential issues when extracting with an active span context (such as with a span link),
+    // the `xray-lambda` propagator SHOULD check if the provided context already has an active span context.
+    // If found, the propagator SHOULD just return the extract result of the `xray` propagator.
     if (Span.fromContext(context).getSpanContext().isValid())
-      return context
+      return xrayPropagator.extract(context, carrier)
+
+    // Update context with the extract result from xray propagator.
+    context = xrayPropagator.extract(context, carrier)
 
-    traceHeader = getEnvironment("_X_AMZN_TRACE_ID");
+    // If xray-lambda environment variable not set, return.
+    traceHeader = getEnvironment("_X_AMZN_TRACE_ID")
     if (isEmptyOrNull(traceHeader))
       return context
 
+    // Apply the xray propagator using the span context contained in the xray-lambda environment variable.
     return xrayPropagator.extract(context, ["_X_AMZN_TRACE_ID": traceHeader])
 }
 ```
 
+*Note:* Java implementations should use the system property value of the key `com.amazonaws.xray.traceHeader`
+instead of the environment variable if the system property is not empty.
+
 #### `xray-lambda` Propagator Configuration
 
-Since propagators are invoked in order, users would give priority to X-Ray's "Active Tracing" span context by setting the environment variable:
+**When reporting spans to AWS X-Ray** from AWS Lambda, the `xray-lambda` propagator SHOULD replace the `xray` propagator in the `OTEL_PROPAGATORS` configuration. Including both will prevent `xray-lambda` from functioning properly.
+
+Example valid configuration when reporting spans to AWS X-Ray:
+
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray-lambda`
+
+Example invalid configurations:
 
-`OTEL_PROPAGATORS=tracecontext,baggage,xray,xray-lambda`
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray,xray-lambda`
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray-lambda,xray`
 
-To avoid broken traces, if OpenTelemetry is reporting traces to another system besides AWS X-Ray, users should either omit `xray-lambda` or add it to the beginning:
+**When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken:
 
-`OTEL_PROPAGATORS=xray-lambda,tracecontext,baggage,xray`
+Example valid configuration when OpenTelemetry is reporting traces to another system besides AWS X-Ray:
 
-*Note: The `xray-lambda` propagator can only `extract` context. The `inject` operation MUST be a no-op.*
+- `OTEL_PROPAGATORS=xray-lambda,tracecontext,baggage,xray`
```

Comment:
Hum above it says:

> **When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken:

Isn't this example here wrong then? 🤔

Reply:
Oops, good catch. Copy paste error.